packages/std/src/math/int64.rs

```diff
@@ -0,0 +1,1047 @@
+    pub fn saturating_mul(self, other: Self) -> Self {
+        Self(self.0.saturating_mul(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_pow(self, exp: u32) -> Self {
+        Self(self.0.saturating_pow(exp))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn abs_diff(self, other: Self) -> Uint64 {
+        Uint64(self.0.abs_diff(other.0))
+    }
+}
+
+impl From<u32> for Int64 {
+    fn from(val: u32) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<u16> for Int64 {
+    fn from(val: u16) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<u8> for Int64 {
+    fn from(val: u8) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<i64> for Int64 {
+    fn from(val: i64) -> Self {
+        Int64(val)
+    }
+}
+
+impl From<i32> for Int64 {
+    fn from(val: i32) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<i16> for Int64 {
+    fn from(val: i16) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl From<i8> for Int64 {
+    fn from(val: i8) -> Self {
+        Int64(val.into())
+    }
+}
+
+impl TryFrom<&str> for Int64 {
+    type Error = StdError;
+
+    fn try_from(val: &str) -> Result<Self, Self::Error> {
+        Self::from_str(val)
+    }
+}
+
+impl FromStr for Int64 {
+    type Err = StdError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.parse::<i64>() {
+            Ok(u) => Ok(Self(u)),
+            Err(e) => Err(StdError::generic_err(format!("Parsing Int64: {}", e))),
+        }
+    }
+}
+
+impl From<Int64> for String {
+    fn from(original: Int64) -> Self {
+        original.to_string()
+    }
+}
+
+impl fmt::Display for Int64 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl Add<Int64> for Int64 {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self {
+        Int64(self.0.checked_add(rhs.0).unwrap())
+    }
+}
+forward_ref_binop!(impl Add, add for Int64, Int64);
+
+impl Sub<Int64> for Int64 {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self {
+        Int64(self.0.checked_sub(rhs.0).unwrap())
+    }
+}
+forward_ref_binop!(impl Sub, sub for Int64, Int64);
+
+impl SubAssign<Int64> for Int64 {
+    fn sub_assign(&mut self, rhs: Int64) {
+        self.0 = self.0.checked_sub(rhs.0).unwrap();
+    }
+}
+forward_ref_op_assign!(impl SubAssign, sub_assign for Int64, Int64);
+
+impl Div<Int64> for Int64 {
+    type Output = Self;
+
+    fn div(self, rhs: Self) -> Self::Output {
+        Self(self.0.checked_div(rhs.0).unwrap())
+    }
+}
+forward_ref_binop!(impl Div, div for Int64, Int64);
+
+impl Rem for Int64 {
+    type Output = Self;
+
+    /// # Panics
+    ///
+    /// This operation will panic if `rhs` is zero.
+    #[inline]
+    fn rem(self, rhs: Self) -> Self {
+        Self(self.0.rem(rhs.0))
```

Comment:
Nit: Rely on `checked_rem` for consistency?

Reply:
I tried to keep it consistent with the unsigned ones (they have the same implementation for rem).

Actually, I think it might be better to change the ones using `self.0.checked_*` to use either `self.0.*`, a `self.checked_*` call or an explicit message using `.expect`. That would give more helpful panic messages.
But I think we can do that later.